src/rating-indicator.scss

```diff
@@ -0,0 +1,146 @@
+@import "./new-settings";
+@import "./mixins";
+@import "./icon";
+
+$block: #{$fd-namespace}-rating-indicator;
+
+.#{$block} {
+  $unratedColor: var(--sapContent_UnratedColor);
+  $fd-rating-indicator-sizes: (
+    "cozy": ("fontSize": 1.5rem, "spacingBetweenIcons": 0.1875rem, "controlMargin": 0.8125rem 0),
+    "compact": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem, "controlMargin": 0.5rem 0),
+    "condensed": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "xs": ("fontSize": 0.75rem, "spacingBetweenIcons": 1px),
+    "s": ("fontSize": 1rem, "spacingBetweenIcons": 0.125rem),
+    "m": ("fontSize": 1.375rem, "spacingBetweenIcons": 0.1875rem),
+    "l": ("fontSize": 2rem, "spacingBetweenIcons": 0.25rem)
+  );
+  $fd-rating-indicator-default-size: map-get($fd-rating-indicator-sizes, 'm');
+
+  @include fd-reset();
+
+  &__container {
+    @include fd-fiori-focus();
+  }
+
+  &__input {
+    @include fd-reset();
+
+    position: absolute;
+    opacity: 0.0001;
+    z-index: -1;
+
+    &:checked ~ input + .#{$block}__label {
+      @include fd-icon-glyph('unfavorite') {
+        color: $unratedColor;
+      }
+
+      &:hover::before {
+        opacity: 1;
+      }
+    }
+  }
+
+  &__label {
+    @include fd-reset();
+    @include action-cursor();
+
+    @include fd-icon-glyph('favorite') {
+      font-family: "SAP-icons";
+      color: var(--sapContent_RatedColor);
+      font-size: map_get($fd-rating-indicator-default-size, "fontSize");
+      margin-right: map_get($fd-rating-indicator-default-size, "spacingBetweenIcons");
+    }
+
+    display: block;
+    float: left;
+
+    @include fd-rtl() {
+      float: right;
+    }
+
+    &:hover::before {
+      opacity: 0.9;
+      transition: opacity $fd-animation-speed ease-in;
+    }
+  }
+
+  &--hide-dynamic-text {
+    > .dynamic-text {
+      display: none;
+    }
+  }
+
+  > .dynamic-text {
+    @include fd-reset();
+    @include fd-ellipsis();
+
+    font-size: var(--sapFontSmallSize);
+    color: var(--sapContent_LabelColor);
+    margin-left: 0.5rem;
+  }
+
+  @include fd-disabled() {
+    pointer-events: none;
+    opacity: var(--sapContent_DisabledOpacity);
+  }
+
+  @include fd-readonly() {
+    pointer-events: none;
+
+    .#{$block}__input {
+      &:checked ~ input + .#{$block}__label {
+        @include fd-icon-glyph('favorite') {
+          color: $unratedColor;
+        }
+      }
+    }
+  }
+
+  @include fd-flex-center();
+
+  &.is-display-mode {
```

Comment:
can't this be a modifier class? Don't have to agree with me :D Just asking for your opinion on this :)

Reply:
I drew attention to the name of the classes for states, I'm about `.is-active`, `.is-disabled`, `.is-focus`, `.is-readonly`.
And `.is-display-mode` it is state but only for rating indicator

Reply:
thank you for the comments 🙂